File: backend/src/main/java/com/medkit/backend/dto/response/AuthResponse.java
package com.medkit.backend.dto.response;

import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
public class AuthResponse {

    private String token;
    private String refreshToken;
    private String type = "Bearer";
    private Integer userId;
    private Integer doctorId;
    private String email;
    private String role;
    private String firstName;
    private String lastName;
    private String middleName;

    public AuthResponse(String token, String refreshToken, Integer userId, String email, String role, String firstName, String lastName) {
        this.token = token;
        this.refreshToken = refreshToken;
        this.userId = userId;
        this.email = email;
        this.role = role;
        this.firstName = firstName;
        this.lastName = lastName;
    }

    public AuthResponse(String token, String refreshToken, Integer userId, Integer doctorId, String email, String role, String firstName, String lastName, String middleName) {
        this.token = token;
        this.refreshToken = refreshToken;
        this.userId = userId;
        this.doctorId = doctorId;
        this.email = email;
        this.role = role;
        this.firstName = firstName;
        this.lastName = lastName;
        this.middleName = middleName;
    }
}

File: medkit_app/lib/data/models/auth_response.dart
class AuthResponse {
  final String token;
  final String? refreshToken;
  final String? type;
  final int userId;
  final int? doctorId;
  final String email;
  final String role;
  final String firstName;
  final String lastName;
  final String? middleName;

  AuthResponse({
    required this.token,
    this.refreshToken,
    this.type,
    required this.userId,
    this.doctorId,
    required this.email,
    required this.role,
    required this.firstName,
    required this.lastName,
    this.middleName,
  });

  factory AuthResponse.fromJson(Map<String, dynamic> json) {
    return AuthResponse(
      token: json['token'] as String,
      refreshToken: json['refreshToken'] as String?,
      type: json['type'] as String?,
      userId: json['userId'] as int,
      doctorId: json['doctorId'] as int?,
      email: json['email'] as String,
      role: json['role'] as String,
      firstName: json['firstName'] as String,
      lastName: json['lastName'] as String,
      middleName: json['middleName'] as String?,
    );
  }
}
